server/service/apple_mdm.go

```diff
@@ -2076,6 +2076,7 @@ func ReconcileProfiles(
 				level.Debug(logger).Log("err", "sending push notifications, profiles still enqueued", "details", err)
 			case err != nil:
 				level.Error(logger).Log("err", "removing profiles from devices", "details", err)
+				// TODO(mna): Am I missing something or are we sending two times for a single host here in case of error?
```

Comment:
Just want to make sure there's not a subtlety I'm missing here, but I think it must return after sending the error on the channel, to avoid sending twice for a given host profile (which would cause the last send to block forever, as the loop further down that receives from the channel would not wait to receive it). We should eventually refactor this a bit to use a range over a channel and close the channel once done sending, which is a safer approach to producer-consumer over a channel.

Reply:
I think that's a bug! what a great catch, want me to open an issue or do you want to fix this here? up to you